modules/library/main/src/main/java/org/geotools/filter/function/FilterFunction_strURLEncode.java

```diff
@@ -0,0 +1,62 @@
+package org.geotools.filter.function;
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.opengis.filter.capability.FunctionName;
+
+/**
+ * URL encodes a string.
+ * This is intended to encode URL paths as such, spaces are encoded as %20, not +
+ *
+ * @see URLEncoder#encode(String)
+ *
+ * @author Billy Newman (BIT Systems)
+ */
+public class FilterFunction_strURLEncode extends FunctionExpressionImpl {
+
+    /**
+     * The FunctionName
+     */
+    public static FunctionName NAME = new FunctionNameImpl("strURLEncode", String.class,
+            parameter("encodeable", String.class));
+
+    /**
+     * Create a new FilterFunction_strURLEncode instance
+     */
+    public FilterFunction_strURLEncode() {
+        super(NAME);
+    }
+
+    /**
+     * URL encode the string.
+     *
+     * @param feature
+     * @return The URL encoded string
+     * @throws IllegalArgumentException if the argument is not a string
+     */
+    @Override
+    public Object evaluate(Object feature) {
+        String arg0;
+
+        try { // attempt to get value and perform encoding
+            arg0 = getExpression(0).evaluate(feature, String.class);
+        } catch (Exception e) // probably a type error
+        {
+            throw new IllegalArgumentException(
+                    "Filter Function problem for function strURLEncode argument #0 - expected type String");
+        }
+
+        try {
+            // Using URLEncoder, spaces are converted to plus signs, but must be %20 for URL calls
+            return URLEncoder.encode(arg0, "utf-8").replaceAll("\\+", "%20");
```

Comment:
Yeah no problem, I was wondering about this and glad you brought it up.  I added an optional    2nd parameter 'formUrlEncode' that defaults to false if not provided.

Thoughts?